OptimizelySDK/Optimizely.cs

```diff
@@ -154,32 +192,66 @@ public static String SDK_TYPE
         /// Initializes a new instance of the <see cref="T:OptimizelySDK.Optimizely"/> class.
         /// </summary>
         /// <param name="configManager">Config manager.</param>
+        /// <param name="notificationCenter">Notification center</param>
         /// <param name="eventDispatcher">Event dispatcher.</param>
         /// <param name="logger">Logger.</param>
         /// <param name="errorHandler">Error handler.</param>
         /// <param name="userProfileService">User profile service.</param>
         /// <param name="eventProcessor">EventProcessor</param>
+        /// <param name="defaultDecideOptions">Default Decide options</param>
+        /// <param name="odpManager">Optional ODP Manager</param>
         public Optimizely(ProjectConfigManager configManager,
-                         NotificationCenter notificationCenter = null,
-                         IEventDispatcher eventDispatcher = null,
-                         ILogger logger = null,
-                         IErrorHandler errorHandler = null,
-                         UserProfileService userProfileService = null,
-                         EventProcessor eventProcessor = null,
-                         OptimizelyDecideOption[] defaultDecideOptions = null)
+            NotificationCenter notificationCenter = null,
+            IEventDispatcher eventDispatcher = null,
+            ILogger logger = null,
+            IErrorHandler errorHandler = null,
+            UserProfileService userProfileService = null,
+            EventProcessor eventProcessor = null,
+            OptimizelyDecideOption[] defaultDecideOptions = null
+#if USE_ODP
+            , OdpManager odpManager = null
+#endif
+        )
         {
             ProjectConfigManager = configManager;
 
-            InitializeComponents(eventDispatcher, logger, errorHandler, userProfileService, notificationCenter, eventProcessor, defaultDecideOptions);
+#if USE_ODP
+            InitializeComponents(eventDispatcher, logger, errorHandler, userProfileService,
+                notificationCenter, eventProcessor, defaultDecideOptions, odpManager);
+
+            var projectConfig = ProjectConfigManager.CachedProjectConfig;
+            if (projectConfig != null)
```

Comment:
The problem with this check is that in case of polling projectconfig would be null and it will not add the Notification listener. So odp setting will never update. Instead of this, change it to this:

`if (ProjectConfigManager.CachedProjectConfig != null)
            {
                // in case Project config is instantly available
                OdpManager?.UpdateSettings(projectConfig.PublicKeyForOdp, projectConfig.HostForOdp,
                    projectConfig.Segments.ToList());
            }
            if (ProjectConfigManager.SdkKey != null)
            {
                NotificationCenterRegistry.GetNotificationCenter(configManager.SdkKey, logger)?.
                AddNotification(NotificationCenter.NotificationType.OptimizelyConfigUpdate,
                    () =>
                    {
                        OdpManager?.UpdateSettings(projectConfig.PublicKeyForOdp,
                            projectConfig.HostForOdp,
                            projectConfig.Segments.ToList());
                    });
            }`